hack/ukify/measure/measure.go

```diff
@@ -0,0 +1,259 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package measure
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/google/go-tpm-tools/simulator"
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+
+	"github.com/siderolabs/ukify/constants"
+)
+
+type PCRData struct {
+	SHA1   []bankData `json:"sha1,omitempty"`
+	SHA256 []bankData `json:"sha256,omitempty"`
+	SHA384 []bankData `json:"sha384,omitempty"`
+	SHA512 []bankData `json:"sha512,omitempty"`
+}
+
+type bankData struct {
+	// list of PCR banks
+	PCRS []int `json:"pcrs"`
+	// Public key of the TPM
+	PKFP string `json:"pkfp"`
+	// Policy digest
+	POL string `json:"pol"`
+	// Signature of the policy digest in base64
+	SIG string `json:"sig"`
+}
+
+// signatureData returns the hashed signature digest and base64 encoded signature
+type signatureData struct {
+	Digest          string
+	SignatureBase64 string
+}
+
+// SectionData holds a map of Section to file path to the corresponding section
+type SectionsData map[constants.Section]string
+
+func calculatePCRBankData(pcr int, alg tpm2.Algorithm, sectionData SectionsData, privateKeyFile string) ([]bankData, error) {
```

Comment:
can we do some unit-test with known data here? to make sure it generates same hashes/signatures always?

Reply:
yes, I'll add a code that actually invokes `sd-measure` and compare output (but that means we need to keep sd-measure in tools if we want to run this in CI? Or have both, one with testdata and expected output and another that can be run locally on a system that has swtpm and sd-measure.